examples/calibration/plot_decision_threshold_calibration.py

```diff
@@ -0,0 +1,167 @@
+"""
+======================================================================
+Decision threshold (cutoff point) calibration on breast cancer dataset
+======================================================================
+
+Machine learning classifiers often base their predictions on real-valued
+decision functions that don't always have accuracy as their objective. Moreover
+the learning objective of a model can differ from the user's needs hence using
+an arbitrary decision threshold as defined by the model can be not ideal.
+
+The CutoffClassifier can be used to calibrate the decision threshold of a model
+in order to increase the classifier's trustworthiness. Optimization objectives
+during the decision threshold calibration can be the true positive and / or
+the true negative rate as well as the f beta score.
+
+In this example the decision threshold calibration is applied on two
+classifiers trained on the breast cancer dataset. The goal in the first case is
+to maximize the f1 score of the classifiers whereas in the second the goal is
+to maximize the true positive rate while maintaining a minimum true negative
+rate.
+
+As you can see after calibration the f1 score of the LogisticRegression
+classifiers has increased slightly whereas the accuracy of the
+AdaBoostClassifier classifier has stayed the same.
+
+For the second goal as seen after calibration both classifiers achieve better
+true positive rate while their respective true negative rates have decreased
+slightly or remained stable.
+"""
+
+# Author: Prokopios Gryllos <prokopis.gryllos@sentiance.com>
+#
+# License: BSD 3 clause
+
+from __future__ import division
+
+import numpy as np
+
+from sklearn.ensemble import AdaBoostClassifier
+from sklearn.metrics import confusion_matrix, f1_score
+from sklearn.calibration import CutoffClassifier
+from sklearn.linear_model import LogisticRegression
+from sklearn.datasets import load_breast_cancer
+import matplotlib.pyplot as plt
+from sklearn.model_selection import train_test_split
+
+
+print(__doc__)
+
+# percentage of the training set that will be used for calibration
+calibration_samples_percentage = 0.2
+
+X, y = load_breast_cancer(return_X_y=True)
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, train_size=0.6,
+                                                    random_state=42)
+
+calibration_samples = int(len(X_train) * calibration_samples_percentage)
+
+lr = LogisticRegression().fit(
```

Comment:
please use lists and loops rather than this repetitive code and hard-to-read variable names like "f_one_lr_f_beta". This should rather be in some record-based structure (dicts? arrays? I don't mind) with strategy='f1', estimator='logistic', f1=value.